src/concrete/ml/onnx/ops_impl.py

```diff
@@ -1495,7 +1496,8 @@ def numpy_cast(data: numpy.ndarray, *, to: int) -> Tuple[numpy.ndarray]:
     if isinstance(data, RawOpOutput):
         return (data.astype(onnx.helper.tensor_dtype_to_np_dtype(to)).view(RawOpOutput),)
 
-    assert_true(to == onnx.TensorProto.BOOL)
+    # Allow both bool and float types
+    assert_true(to in (onnx.TensorProto.BOOL, onnx.TensorProto.FLOAT))
```

Comment:
yes not sure where this is used or tested !

Reply:
but this test itself is meant for another feature. Could you maybe add a specific test that only does this `to(float)` ?

Reply:
I have a test that check different cast yes.

Reply:
ah yes sorry didn't see, great then thanks !